core/src/main/java/io/kestra/core/tasks/flows/ForEachItem.java

```diff
@@ -239,135 +241,300 @@ public class ForEachItem extends Task implements ExecutableTask<ForEachItem.Outp
+    }
+
+    @Hidden
+    @Getter
+    @NoArgsConstructor
+    public static class ForEachItemSplit extends Task implements RunnableTask<ForEachItemSplit.Output> {
+        private String items;
+        private Batch batch;
+
+        private ForEachItemSplit(String parentId, String items, Batch batch) {
+            this.items = items;
+            this.batch = batch;
+
+            this.id = parentId + "_split";
+            this.type = ForEachItemSplit.class.getName();
         }
 
-        try {
+        @Override
+        public ForEachItemSplit.Output run(RunContext runContext) throws Exception {
+            var renderedUri = runContext.render(this.items);
+            if (!renderedUri.startsWith("kestra://")) {
+                var errorMessage = "Unable to split the items from " + renderedUri + ", this is not an internal storage URI!";
+                runContext.logger().error(errorMessage);
+                throw new IllegalArgumentException(errorMessage);
+            }
+
             List<URI> splits = StorageService.split(runContext, this.batch, URI.create(renderedUri));
+            String fileContent = splits.stream().map(uri -> uri.toString()).collect(Collectors.joining(System.lineSeparator()));
+            try (ByteArrayInputStream bis = new ByteArrayInputStream(fileContent.getBytes())){
+                URI splitsFile = runContext.storage().putFile(bis, "splits.txt");
+                return Output.builder().splits(splitsFile).build();
+            }
+        }
 
-            AtomicInteger currentIteration = new AtomicInteger(1);
-
-            return splits.stream()
-                .<SubflowExecution<?>>map(throwFunction(
-                    split -> {
-                        int iteration = currentIteration.getAndIncrement();
-                        // these are special variable that can be passed to the subflow
-                        Map<String, Object> itemsVariable = Map.of("taskrun",
-                            Map.of("items", split.toString(), "iteration", iteration));
-                        Map<String, Object> inputs = new HashMap<>();
-                        if (this.inputs != null) {
-                            inputs.putAll(runContext.render(this.inputs, itemsVariable));
-                        }
+        @Builder
+        @Getter
+        public static class Output implements io.kestra.core.models.tasks.Output {
+            private URI splits;
+        }
+    }
 
-                        List<Label> labels = new ArrayList<>();
-                        if (this.inheritLabels && currentExecution.getLabels() != null && !currentExecution.getLabels().isEmpty()) {
-                            labels.addAll(currentExecution.getLabels());
-                        }
+    @Hidden
+    @Getter
+    @NoArgsConstructor
+    public static class ForEachItemExecutable extends Task implements ExecutableTask<Output> {
```

Comment:
For each new class, dont forget to add the icon (the same as the basic ForEach)

Reply:
The icon is shown on the topology right?

Reply:
The icon is only shown for the ForEachItem task not its 3 subtasks because they are hidden to avoid listing them in the documentation.

Reply:
also on the topology ?

Reply:
Yes, the same for the topology, as the subtasks are hidden there is no icon.